mycroft/skills/intent_service.py

```diff
@@ -443,14 +436,28 @@ def handle_get_intent(self, message):
         utterance = message.data["utterance"]
         lang = message.data.get("lang", "en-us")
         combined = _normalize_all_utterances([utterance])
-        adapt_intent = self.adapt_service.match_intent(combined, lang)
-        # Adapt intent's handler is used unless
-        # Padatious is REALLY sure it was directed at it instead.
-        padatious_intent = self.padatious_service.match_high(combined)
-        intent = padatious_intent or adapt_intent
-        intent_data = intent.intent_data if intent else None
-        self.bus.emit(message.reply("intent.service.intent.reply",
-                                    {"intent": intent_data}))
+
+        # List of functions to use to match the utterance with intent.
+        # These are listed in priority order.
+        match_funcs = [
+            self.padatious_service.match_high,
+            self.adapt_service.match_intent, self.fallback.high_prio,
+            self.padatious_service.match_medium, self.fallback.medium_prio,
+            self.padatious_service.match_low, self.fallback.low_prio
+        ]
+        # Loop through the matching functions until a match is found.
+        for match_func in match_funcs:
+            match = match_func(combined, lang, message)
+            if match:
+                if match.intent_type:
+                    intent_data = match.intent_data
+                    intent_data["intent_name"] = match.intent_type
+                    intent_data["intent_service"] = match.intent_service
+                    intent_data["skill_id"] = match.skill_id
+                    intent_data["handler"] = match_func.__name__
+                    self.bus.emit(message.reply("intent.service.intent.reply",
+                                                {"intent": intent_data}))
+                break
```

Comment:
Needs a
```python
        else:                                                                 
              self.bus.emit(message.reply("intent.service.intent.reply",        
                                          {"intent": None}))
```

In case no match at all is found.